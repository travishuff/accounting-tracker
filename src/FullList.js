import React from 'react'
import axios from 'axios'
import { Link } from '@reach/router'
import sortBy from 'lodash/sortBy'

import { Container, Table } from 'semantic-ui-react'

class FullList extends React.Component {
  state = {
    column: null,
    direction: null,
    bananas: [],
  }

  handleSort = clickedColumn => () => {
    const { column, bananas, direction } = this.state

    if (column !== clickedColumn) {
      this.setState({
        column: clickedColumn,
        bananas: sortBy(bananas, [clickedColumn]),
        direction: 'ascending',
      })

      return
    }

    this.setState({
      bananas: bananas.reverse(),
      direction: direction === 'ascending' ? 'descending' : 'ascending',
    })
  }

  async componentDidMount() {
    const response = await axios
      .get('http://localhost:8080/api/bananas')
      .catch(console.error)
    this.setState({ bananas: response.data })
  }

  render() {
    const { column, direction, bananas } = this.state

    return (
      <Container fluid={true} className="main-container">
        <h1>
          <Link to="/analytics">‚Üê</Link> Full List of Bananas
        </h1>

        <Table color="green" sortable celled unstackable>
          <Table.Header>
            <Table.Row>
              <Table.HeaderCell
                sorted={column === 'id' ? direction : null}
                onClick={this.handleSort('id')}
              >
                ID ({bananas.length} total)
              </Table.HeaderCell>
              <Table.HeaderCell
                sorted={column === 'buyDate' ? direction : null}
                onClick={this.handleSort('buyDate')}
              >
                buyDate
              </Table.HeaderCell>
              <Table.HeaderCell
                sorted={column === 'sellDate' ? direction : null}
                onClick={this.handleSort('sellDate')}
              >
                sellDate
              </Table.HeaderCell>
            </Table.Row>
          </Table.Header>
          <Table.Body>
            {bananas.map(({ buyDate, sellDate, id }) => (
              <Table.Row key={id}>
                <Table.Cell>{id}</Table.Cell>
                <Table.Cell>{buyDate}</Table.Cell>
                <Table.Cell>{sellDate === null ? 'null' : sellDate}</Table.Cell>
              </Table.Row>
            ))}
          </Table.Body>
        </Table>
      </Container>
    )
  }
}

export default FullList
