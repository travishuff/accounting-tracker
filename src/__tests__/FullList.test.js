import React from 'react'
import { create } from 'react-test-renderer'

import FullList from '../FullList'

// Snapshot test
test('snapshot', () => {
  const c = create(<FullList />)

  expect(c.toJSON()).toMatchSnapshot()
})
