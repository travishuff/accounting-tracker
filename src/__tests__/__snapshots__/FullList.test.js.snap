// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="ui fluid container main-container"
>
  <h1>
    <a
      href="/analytics"
      onClick={[Function]}
    >
      ‚Üê
    </a>
     Full List of Bananas
  </h1>
  <table
    className="ui green celled sortable unstackable table"
  >
    <thead
      className=""
    >
      <tr
        className=""
      >
        <th
          className=""
          onClick={[Function]}
        >
          ID (
          0
           total)
        </th>
        <th
          className=""
          onClick={[Function]}
        >
          buyDate
        </th>
        <th
          className=""
          onClick={[Function]}
        >
          sellDate
        </th>
      </tr>
    </thead>
    <tbody
      className=""
    />
  </table>
</div>
`;
